.gitignore

```diff
@@ -1,7 +1,28 @@
+# OS Specific
 .DS_Store
+.DS_Store?
+._*
+.Spotlight-V100
+.Trashes
+ehthumbs.db
+Thumbs.db
+
+# IDE Specific
+nbproject
+.~lock.*
+.buildpath
+.idea
+composer.lock
+*.sublime-workspace
+*.swp
+*.swo
+/.settings/
+/.project
```

Comment:
If we go the route of not adding environment specific ignores to the project, can we please add something in CONTRIBUTING.md to reflect project policy & modus operandi with global gitignore.